core/domain/suggestion_jobs_one_off.py

```diff
@@ -236,3 +238,122 @@ def map(item):
     @staticmethod
     def reduce(key, values):
         yield (key, len(values))
+
+
+class PopulateCommunityContributionStatsOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """A reusable one-time job that may be used to initialize, or regenerate,
+    the reviewer and suggestion counts in the CommunityContributionStatsModel.
+    """
+
+    _VALIDATION_ERROR_KEY = 'community_contribution_stats_validation_error'
+    ITEM_TYPE_REVIEWER = 'reviewer'
+    ITEM_TYPE_SUGGESTION = 'suggestion'
+    ITEM_CATEGORY_QUESTION = 'question'
+    ITEM_CATEGORY_TRANSLATION = 'translation'
+    KEY_DELIMITER = '.'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [
+            suggestion_models.GeneralSuggestionModel,
+            user_models.UserContributionRightsModel
+        ]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        if isinstance(item, suggestion_models.GeneralSuggestionModel):
+            # Exit early if the suggestion type is not a part of the
+            # Contributor Dashboard or if the suggestion is not currently in
+            # review.
+            if item.suggestion_type == (
+                    suggestion_models.SUGGESTION_TYPE_EDIT_STATE_CONTENT) or (
+                        item.status != suggestion_models.STATUS_IN_REVIEW):
+                return
+            suggestion = suggestion_services.get_suggestion_from_model(item)
+            yield ('%s%s%s%s%s' % (
+                (
+                    PopulateCommunityContributionStatsOneOffJob
```

Comment:
The name could just be PopulateContributionStatsOneOffJob if you want shorter naming.

Reply:
Thanks - made the change!